mcs/class/System/System.Net/ServicePoint.cs

```diff
@@ -314,7 +314,7 @@ bool CheckAvailableForRecycling (out DateTime outIdleSince)
 
 				if (removeList != null) {
 					foreach (var group in removeList)
-						if (groups.ContainsKey (group.Name))
+						if (groups != null && groups.ContainsKey (group.Name))
```

Comment:
Thanks. I had thought about that when initially looking at the code, but must have forgotten by the time I got the code pulled. Fixed in c90f9a9